src/TRDH_alg.jl

```diff
@@ -105,266 +190,341 @@ function TRDH(
   selected::AbstractVector{<:Integer} = 1:length(x0),
   kwargs...,
 ) where {R <: Real, F, G, H, DQN <: AbstractDiagonalQuasiNewtonOperator, X}
-  start_time = time()
-  elapsed_time = 0.0
-  ϵ = options.ϵa
-  ϵr = options.ϵr
-  Δk = options.Δk
-  neg_tol = options.neg_tol
-  verbose = options.verbose
-  maxIter = options.maxIter
-  maxTime = options.maxTime
-  η1 = options.η1
-  η2 = options.η2
-  γ = options.γ
-  α = options.α
-  β = options.β
-  reduce_TR = options.reduce_TR
-
-  local l_bound, u_bound
-  has_bnds = false
-  kw_keys = keys(kwargs)
-  if :l_bound in kw_keys
-    l_bound = kwargs[:l_bound]
-    has_bnds = has_bnds || any(l_bound .!= R(-Inf))
-  end
-  if :u_bound in kw_keys
-    u_bound = kwargs[:u_bound]
-    has_bnds = has_bnds || any(u_bound .!= R(Inf))
-  end
+  nlp = FirstOrderModel(f, ∇f!, x0)
```

Comment:
`FirstOrderModel` no longer exists: https://github.com/JuliaSmoothOptimizers/RegularizedProblems.jl/blob/main/src/types.jl#L4

We show now use ManualNLPModels.jl.